### PR TITLE
Update: change supported images

### PR DESCRIPTION
**Pull request by BenjamenMeyer**
Add Distros
- Ubuntu (Noble)
- Linux Mint (21.2, 21.3, 22)
- OpenSuSe (15.6)
- Rocky Linux (8.10, 9.4)

Drop Distros
- Ubuntu (mantic, bionic)
- Debian (buster)
- Fedora (34, 35, 36, 37, 38)
- Rocky Linux (8.5, 8.6, 8.7, 8.8, 8.9, 9.0, 9.1, 9.2, 9.3)

### CODE REVIEW

**Review by royfalk — APPROVED** [review]
Looks fine. How do we test these new versions? Are we using github CICD to check they compile?

File: .github/workflows/gh-actions-pr.yml [hunk 1]
```diff
@@ -8,39 +8,30 @@ jobs:
...
           - 'debian:bullseye'
-          #- 'debian:buster'        # buster-backports is no longer available from deb.debian.org. Sunsetting?
+          - 'opensuse/leap:15.6'
           - 'opensuse/leap:15.5'
           - 'opensuse/leap:15.4'
           - 'opensuse/leap:15.3'
           - 'opensuse/leap:15.2'
+          #- 'fedora:41'
           - 'fedora:40'
           - 'fedora:39'
```
[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply .github/workflows/gh-actions-pr.yml:32]: These directly translate to Docker Image names so no we can't use comparisons, etc.

File: script/bootstrap [hunk 2]
```diff
@@ -124,16 +124,43 @@ function bootstrapOnDebian()
                       lsb-release
       ;;
     "buster")
+      echo "Sorry, Debian buster is no longer supported"
```
[the remark this answers is not in the document]
Comment (BenjamenMeyer) [reply script/bootstrap:127]: Its more for helping us catch what we deprecated. We can clean them up to reduce the list after a time.

**Comment by BenjamenMeyer** [comment]
> Looks fine. How do we test these new versions? Are we using github CICD to check they compile?

We use GitHub Actions to see that the images build successfully. Generally this works fine, but when there's a major distro change we might have to rework dependencies but we won't catch that until we're using the images in the engine repo right now.

Not sure we need to check that in the GHA here. I have a modified version of one of the scripts for doing local image builds to test and help diagnose failures there. I could probably add a second for running with the engine build too as a local test thing.

We do want to keep the effort here relatively light.

**Pull request merged**